pkg/dao/dao.go

```diff
@@ -27,7 +23,7 @@ type Dao struct {
 	config    Config
```

Comment:
Thinking about it more, the app.go code should delegate the validation of a data connection to the dao and the dao determines what that means since it really belongs in the domain of that package. A good indicator of that abstraction leak is if other code in the broker is importing etcd/client.